src/pages/Settings/NetworkSettingsSection.tsx

```diff
@@ -0,0 +1,148 @@
+// Copyright 2018 - 2021 The Alephium Authors
+// This file is part of the alephium project.
+//
+// The library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the library. If not, see <http://www.gnu.org/licenses/>.
+
+import { AlertTriangle } from 'lucide-react'
+import { useContext, useState, useCallback } from 'react'
+import styled from 'styled-components'
+
+import { GlobalContext } from '../../App'
+import { Button } from '../../components/Buttons'
+import ExpandableSection from '../../components/ExpandableSection'
+import InfoBox from '../../components/InfoBox'
+import { Select, Input } from '../../components/Inputs'
+import { Section } from '../../components/PageComponents/PageContainers'
+import { useMountEffect } from '../../utils/hooks'
+import { NetworkType, getNetworkName, networkEndpoints, Settings } from '../../utils/settings'
+
+interface NetworkSelectOption {
+  label: string
+  value: NetworkType
+}
+
+const NetworkSettingsSection = () => {
+  const { settings: currentSettings, setSettings, setSnackbarMessage } = useContext(GlobalContext)
+
+  const [tempSettings, setTempSettings] = useState<Settings>({
+    nodeHost: currentSettings.nodeHost,
+    explorerApiHost: currentSettings.explorerApiHost,
+    explorerUrl: currentSettings.explorerUrl
+  })
+
+  const [selectedNetwork, setSelectedNetwork] = useState<NetworkType>()
+
+  const [advancedSectionOpen, setAdvancedSectionOpen] = useState(false)
+
+  const networkSelectOptions: NetworkSelectOption[] = [
+    { label: 'Mainnet', value: 'mainnet' },
+    { label: 'Testnet', value: 'testnet' },
+    { label: 'Localhost', value: 'localhost' },
+    { label: 'Custom', value: 'custom' }
+  ]
+
+  const overrideSelectionIfMatchesPreset = useCallback((newSettings: Settings) => {
+    // Check if values correspond to an existing preset
+    const newNetwork = getNetworkName(newSettings)
+
+    setSelectedNetwork(newNetwork)
+  }, [])
+
+  const editSettings = (v: Partial<Settings>) => {
+    const newSettings = { ...tempSettings, ...v }
+
+    // Check if we need to append the http:// protocol if an IP or localhost is used
+    if (v.nodeHost?.match(/^(?:(?:[0-9]{1,3}\.){3}[0-9]{1,3}|localhost)(?::[0-9]*)?$/)) {
+      newSettings.nodeHost = `http://${v.nodeHost}`
```

Comment:
Is there a reason why we use `http` instead of `https`?

Reply:
I believe we have to for localhost (no certificate used locally). 
As for testnet and mainnet, I think the protocol is automatically upgraded (at least with our nodes).
We could append `http` for localhost, `https` for others.